configtype/duration_test.go

```diff
@@ -52,3 +56,74 @@ func TestComparability(t *testing.T) {
 		}
 	}
 }
+
+func TestDuration_JSONSchema(t *testing.T) {
+	sc := (&jsonschema.Reflector{RequiredFromJSONSchemaTags: true}).Reflect(configtype.Duration{})
+	schema, err := json.MarshalIndent(sc, "", "  ")
+	require.NoError(t, err)
+
+	validator, err := plugin.JSONSchemaValidator(string(schema))
+	require.NoError(t, err)
+
+	type testCase struct {
+		Name string
+		Spec string
+		Err  bool
+	}
+
+	for _, tc := range append([]testCase{
+		{
+			Name: "empty",
+			Err:  true,
+			Spec: `""`,
+		},
+		{
+			Name: "null",
+			Err:  true,
+			Spec: `null`,
+		},
+		{
+			Name: "bad type",
+			Err:  true,
+			Spec: `false`,
+		},
+		{
+			Name: "bad format",
+			Err:  true,
+			Spec: `false`,
+		},
+	},
+		func() []testCase {
```

Comment:
nit: this syntax of appending a function that calls itself and appends the results in-line feels unnecessarily complicated, it'd be more readable to introduce a variable first and append that, like:

```
generatedCases := make([]testcase, 20)
for i := 0; i < 20; i++ {
     generatedCases[i] = generateTestCase()
}

...
cases := append([]testCase{
		{
			Name: "empty",
			Err:  true,
			Spec: `""`,
		},
                ...,
                generatedCases...
}
```

or have a few manually created test cases, there doesn't seem to be enough complexity here to warrant generating random cases